src/Cli/dotnet/Commands/Run/VirtualProjectBuildingCommand.cs

```diff
@@ -353,7 +353,9 @@ public override int Execute()
                     Debug.Assert(buildRequest.ProjectInstance != null);
 
                     // Cache run info (to avoid re-evaluating the project instance).
-                    cache.CurrentEntry.Run = RunProperties.FromProject(buildRequest.ProjectInstance);
+                    cache.CurrentEntry.Run = RunProperties.TryFromProject(buildRequest.ProjectInstance, out var runProperties)
```

Comment:
Does file-based support multi TFMs?
This is probably a case where the "outer" project evaluation will have a null RunCommand as its value depends on the specific TFM. Does the caching play nice with this scenario somehow? Or is the scenario simply not supported altogether?

Reply:
`dotnet run` doesn't support multi TFMs in general (neither for project-based nor for file-based apps). For file-based libraries (i.e., `dotnet build` only), multi TFM should be supported; and now with this change, cached run properties should not be used for them (since there is no single `TargetFramework`, the `TryFromProject` call will fail and save `null` to the cache). I can try adding some tests, thanks.

Reply:
Sure, but then you basically turn it into a single TFM app

Reply:
If the caching mechanism can already detect whether or not the current run is using the same TFM, then yeah sure.